src/cli/commands/__tests__/lint.test.ts

```diff
@@ -1,448 +1,172 @@
+    },
+  ];
+
+  beforeEach(() => {
+    (lint as jest.Mock).mockClear();
+    (lint as jest.Mock).mockResolvedValueOnce(results);
+
+    (formatOutput as jest.Mock).mockClear();
+    (formatOutput as jest.Mock).mockReturnValueOnce('<formatted output>');
+
+    (writeOutput as jest.Mock).mockClear();
+    (writeOutput as jest.Mock).mockResolvedValueOnce(undefined);
+
+    errorSpy = jest.spyOn(console, 'error');
+  });
 
-      test
-        .stdout()
-        .command(['lint', validOas3SpecPath, '-r', invalidRulesetPath])
-        .exit(2)
-        .it('outputs "invalid ruleset" error', ctx => {
-          expect(ctx.stdout).toContain(`/rules/rule-without-given-nor-them should have required property 'given'`);
-          expect(ctx.stdout).toContain(`/rules/rule-without-given-nor-them should have required property 'then'`);
-          expect(ctx.stdout).toContain(`/rules/rule-with-invalid-enum/severity should be number`);
-          expect(ctx.stdout).toContain(
-            `/rules/rule-with-invalid-enum/severity should be equal to one of the allowed values`,
-          );
-        });
+  afterEach(() => {
+    errorSpy.mockRestore();
+  });
 
-      test
-        .stdout()
-        .command(['lint', validCustomOas3SpecPath, '-r', validRulesetPath])
-        .it('outputs no issues', ctx => {
-          expect(ctx.stdout).toContain('No errors or warnings found!');
-        });
+  it('shows help when no document argument is passed', async () => {
+    const output = await run('lint');
+    expect(output).toContain('document  Location of a JSON/YAML document');
+  });
 
-      test
-        .stdout()
-        .command(['lint', validOas3SpecPath, '-r', validRulesetPath])
-        .it('outputs warnings in default format', ctx => {
-          expect(ctx.stdout).toContain('5:10  warning  info-matches-stoplight  Info must contain Stoplight');
-          expect(ctx.stdout).not.toContain('Info object should contain `contact` object');
-          expect(ctx.stdout).toContain('OpenAPI 3.x detected');
-        });
+  it('shows help when invalid arguments are passed', async () => {
+    const output = await run('lint --foo');
+    expect(output).toContain('lint a JSON/YAML document from a file or URL');
+  });
 
-      test
-        .nock('http://foo.local', api =>
-          api.get('/ruleset.yaml').replyWithFile(200, validRulesetPath, {
-            'Content-Type': 'application/yaml',
-          }),
-        )
-        .stdout()
-        .command(['lint', validCustomOas3SpecPath, '-r', 'http://foo.local/ruleset.yaml'])
-        .it('given valid remote ruleset file, outputs no issues', ctx => {
-          expect(ctx.stdout).toContain('No errors or warnings found!');
-        });
-    });
+  it('calls lint with document and default options', async () => {
+    const doc = './__fixtures__/empty-oas2-document.json';
+    await run(`lint ${doc}`);
+    expect(lint).toBeCalledWith(
+      doc,
+      expect.objectContaining({
```

Comment:
`calls lint with document and default options`, so there's `doc`, so I know this is the document. But not sure what default options are? If default options is the object containing `.encoding` and `.format`, then what is `undefined`? @philsturgeon, do you think that we can use vars (like you did for `doc`) to name these things?

Reply:
I dunno @P0lip wrote these tests :D

Reply:
These tests have been ported :D. Eight way, the defaults in this particular case means no options specified. Its up to the command definition what those defaults are.

Reply:
In general, all options are passed. I'll explicitly pass them here rather than just a subset of them.